src/clients/go/tb_client.go

```diff
@@ -286,211 +278,204 @@ func onGoPacketCompletion(
 		}
 
 		// Write the result data into the request's result.
-		if req.result != nil {
-			wrote = result_len
-			C.memcpy(req.result, unsafe.Pointer(result_ptr), C.size_t(result_len))
-		}
+		reply = make([]uint8, result_len)
+		C.memcpy(unsafe.Pointer(&reply[0]), unsafe.Pointer(result_ptr), C.size_t(result_len))
 	}
 
 	// Signal to the goroutine which owns this request that it's ready.
-	packet.data_size = wrote
-	req.ready <- struct{}{}
+	req.ready <- reply
 }
 
 func (c *c_client) CreateAccounts(accounts []types.Account) ([]types.AccountEventResult, error) {
 	count := len(accounts)
-	results := make([]types.AccountEventResult, count)
-
-	var dataPtr, resultPtr unsafe.Pointer
+	var dataPtr unsafe.Pointer
 	if count > 0 {
 		dataPtr = unsafe.Pointer(&accounts[0])
-		resultPtr = unsafe.Pointer(&results[0])
 	} else {
 		dataPtr = nil
-		resultPtr = nil
 	}
 
-	wrote, err := c.doRequest(
+	reply, err := c.doRequest(
 		C.TB_OPERATION_CREATE_ACCOUNTS,
 		count,
 		dataPtr,
-		resultPtr,
 	)
 
 	if err != nil {
 		return nil, err
 	}
 
-	resultCount := wrote / int(unsafe.Sizeof(types.AccountEventResult{}))
-	return results[0:resultCount], nil
+	if reply == nil {
+		return make([]types.AccountEventResult, 0), nil
+	}
+
+	resultsCount := len(reply) / int(unsafe.Sizeof(types.AccountEventResult{}))
+	results := unsafe.Slice((*types.AccountEventResult)(unsafe.Pointer(&reply[0])), resultsCount)
```

Comment:
Hm, is it valid to cast `[]byte` to `[]T` in Go that way? I can't find a definitive answer one way or another, but what gives me a pause is that this package:

https://github.com/alecthomas/unsafeslice/blob/48ea3b1ba1c124a29aa74f9762f984f74c1533a6/unsafeslice.go#L20-L31

does things quite differently (modifying the actual slice header). 

How can we verify that this cast is indeed valid, and won't confuse garbage collector or cause some other nastiness?

Reply:
Ahh, apparently it _is_ the correct way to do this, it's just relatively new: https://stackoverflow.com/a/69676107

Ok, yeah, if this passes our tests without panicking runtime, I am reasonably sure it is correct!

Reply:
Yes, the slice conversion is fine. However, I'm not ashamed to admit that I don't fully understand how Go's memory management works, and I'm always uncertain whether the GC will interpret our intent correctly, especially in edge cases like this one, where the buffer is allocated from an exported Go function called from Zig!

AFAIK, the GC will track the memory addresses, regardless of the type.
Like you said, "I am reasonably sure it is correct"!